source/MaterialXGenShader/ShaderNode.cpp

```diff
@@ -238,7 +238,7 @@ ShaderNodePtr ShaderNode::create(const ShaderGraph* parent, const string& name,
         {
             newNode->addOutput(port->getName(), portType);
         }
-        else
+        else if (port->isA<Input>())
```

Comment:
This is a regression. 
Tokens are value elements and would be added erroneously as a shader node input port.
This resulted in a shader function with token argument on it which is incorrect.